calicoctl/commands/datastore/migrate/migrateipam.go

```diff
@@ -120,6 +128,38 @@ func (m *migrateIPAM) PullFromDatastore() error {
 		if !ok {
 			return fmt.Errorf("Could not convert %+v to an AllocationBlock", item.Value)
 		}
+
+		// Update node names in the block to match the Kubernetes node
+		if m.nodeMap != nil {
+			for i, allocationAttribute := range block.Attributes {
+				// Update the node name if it has a corresponding Kubernetes node name
+				if nodeName, ok := m.nodeMap[allocationAttribute.AttrSecondary["node"]]; ok {
+					block.Attributes[i].AttrSecondary["node"] = nodeName
+				}
+
+				// Update the handle ID for ipip tunnel addresses
+				if allocationAttribute.AttrPrimary != nil {
+					handlePrefixReplaced := false
```

Comment:
Why do we want this variable here? 

IIUC, it will make sure that we only replace one tunnel address handle and skip any subsequent ones. Is that right? 

I think if we have multiple tunnel addresses present we should replace all of them.

Reply:
Yup, it's for skipping once one is found. I thought we could only have one tunnel so I added it, if we can have multiple (which it sounds like) then this is wrong and I need to remove it.